app/src/main/java/com/jerboa/ui/components/post/PostActivity.kt

```diff
@@ -300,6 +305,19 @@ fun PostActivity(
                     is ApiState.Failure -> ApiErrorText(postRes.msg)
                     is ApiState.Success -> {
                         val postView = postRes.data.post_view
+                        account?.also { _ ->
+                            if (!postView.read) {
+                                appState.addReturn(PostViewReturn.POST_VIEW, postView.copy(read = true))
```

Comment:
This works for this ig, but I will probably change this later to behaviour i described before so that it will also update the saved , likes, etc